### PR TITLE
support reading signing passphrase from file or stdin

### LINKED ISSUE #8210: Use of HELM_KEY_PASSPHRASE to pass passphrase to helm package
In Helm v2, we had the ability to set an environmental variable to pass in the passphrase to the GPG private key using $HELM_KEY_PASSPHRASE. (Added in [pull 4778](https://github.com/helm/helm/pull/4778/files/b0a32d089e5fa91594cdba8b62ecae8ed5294751))

In Helm v3, that feature does not appear to exist. The only passphraseFetcher implemented is in [helm/pkg/action/package.go](https://github.com/helm/helm/blob/d0422e6d82b9d7ddc9c24f02ed1f7d321f9e2646/pkg/action/package.go) on line 136.

Without this option, we can't make use of `helm package --sign` from CI/CD tools (like Azure DevOps, which I'm currently using), because the `terminal.ReadPassword` does not take place in a TTY, which results in an error `inappropriate ioctl for device`.

I'm open to other options as well.

### PR DESCRIPTION
**Pull request by sdorra**
Signed-off-by: Sebastian Sdorra <sebastian.sdorra@cloudogu.com>

closes #8210 

**What this PR does / why we need it**:

In helm v2 we had the HELM_KEY_PASSPHRASE environment variable to pass the passphrase within a ci process, but in helm v3 we have no way to pass the passphrase without user interaction.

So this pr will add support for reading the signing passphrase from file or stdin e.g.:

```bash
echo secret > secret.txt
helm package --sign --key mykey --keyring secring.gpg --passphrase-file secret.txt my-chart
# or
echo secret | helm package --sign --key mykey --keyring secring.gpg --passphrase-file "-" my-chart
```

**If applicable**:
- [ ] this PR contains documentation
- [X] this PR contains unit tests
- [X] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by bacongobbler — APPROVED** [review]
code LGTM, though it'll need another maintainer to approve before merging.

**Comment by sdorra** [comment]
@bacongobbler Is there something i can do to speed things up?

**Comment by Lukkie** [comment]
I also need this for my helm chart build pipeline. Would be nice if it could be merged sometime soon.

**Comment by fcrespofastly** [comment]
Any chance this gets reviewed by someone else @bacongobbler ? We're willing to use this in our build pipelines too

**Comment by bacongobbler** [comment]
Feel free to reach out to the other core maintainers.

**Comment by dkulchinsky** [comment]
Thanks for replying @bacongobbler! but the list of maintainers in OWNERS is quite long, not sure who we should tag here? could you help us figure out who could help us here?

We desperately need this feature for our CI pipeline since we need to sign the charts to generate a provenance file.

**Comment by sdorra** [comment]
It looks like a large part of the GPG code was written by technosophos.

**Comment by sdorra** [comment]
Hey @technosophos could you help us with this pr?

**Comment by technosophos** [comment]
I will be reviewing it today or tomorrow.

**Review by technosophos — APPROVED** [review]
LGTM

**Comment by technosophos** [comment]
Added this to 3.4.0 since it is a new feature.

**Pull request merged**
